cockatrice/src/window_main.cpp

```diff
@@ -288,6 +293,87 @@ void MainWindow::actAbout()
     ));
 }
 
+void MainWindow::actUpdate()
+{
+    //Request the newest builds
+    QUrl versionUrl(LATEST_FILES_URL);
+    response = netMan.get(QNetworkRequest(versionUrl));
+    connect(response, SIGNAL(finished()),
+            this, SLOT(fileListFinished()));
+}
+
+void MainWindow::fileListFinished() {
+    //Parse the version results into a JSON array
+    QString strReply = (QString) response->readAll();
+    QJsonDocument jsonResponse = QJsonDocument::fromJson(strReply.toUtf8());
+    QJsonArray json_array = jsonResponse.array();
+
+    //From this array we intend to find the correct downloadUrl, if needed
+    QString downloadUrl;
+    bool correctVersion = false, needToUpdate = false;
+    QString dateStr;
+    int downloadSize;
+
+    //Iterate over the builds
+    foreach (const QJsonValue &value, json_array)
```

Comment:
Also, instead of inlining the logic, you could have a static method of  `someType parseVersionMetadata(&QJsonValue)` and then `std::transform` the QJsonArray using that function. Having small composable functions like that makes the code easier to reason about and will make unit testing simpler.